### PR TITLE
#24 Resolved support for analyzers which are running only once

### PR DESCRIPTION
**Pull request by mbocek**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,3 +57,5 @@ Configuration for DBadvisor is provided by environment. The list of all environm
 | dbadvisor.report.location                    | _temporary folder, see_ `java.io.tmpdir` _system property_ |
 | dbadvisor.report.on-shutdown-hook            | true          |
 | dbadvisor.max-issues-count-per-type          | 100           |
+| dbadvisor.oracle.foreign-key.enabled         | true          |
+| dbadvisor.oracle.foreign-key.threshold       | 1             |
```
Comment (thradec) [inline README.md]: minor: we should keep this properties together with the rest of "oracle" properties

File: dbadvisor-core/src/main/java/com/github/embeditcz/dbadvisor/core/internal/analyzer/oracle/ForeignKeyAnalayzer.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import java.sql.SQLException;
+import java.util.List;
+import java.util.Map;
+
+import static java.lang.String.format;
+
+@Component
+public class ForeignKeyAnalayzer extends AbstractDatabaseAnalyzer {
+
+    private static String FOREIGN_KEY_QUERY = "WITH constr AS (" +
```
Comment (thradec) [inline dbadvisor-core/src/main/java/com/github/embeditcz/dbadvisor/core/internal/analyzer/oracle/ForeignKeyAnalayzer.java]: minor: final

File: dbadvisor-core/src/main/java/com/github/embeditcz/dbadvisor/core/internal/analyzer/oracle/ForeignKeyAnalayzer.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    @Override
+    protected void analyzeImpl(DatabaseContext ctx) {
+        JdbcTemplate jdbcTemplate = new JdbcTemplate(ctx.getDataSource());
+        String schemaName = resolveSchemaName(ctx.getDataSource());
+        List<Map<String, Object>> foreignKeys = jdbcTemplate.queryForList(FOREIGN_KEY_QUERY, schemaName, schemaName);
+        if (foreignKeys.size() > 0) {
+            Issue issue = issueBuilder.builder()
+                .type(getName())
+                .weight(foreignKeys.size())
+                .query(FOREIGN_KEY_QUERY)
```
Comment (thradec) [inline dbadvisor-core/src/main/java/com/github/embeditcz/dbadvisor/core/internal/analyzer/oracle/ForeignKeyAnalayzer.java]: as we spoke on monday, there is a theoretical issue with discarding issues, which have the same query and lower weight

one posible solution is to use null here and just fix the NPE in findIssueWithIdenticalQuery (which should return null, when the query is null) 

wdyt?

File: README.md (hunk 1 above)
Comment (mbocek) [reply README.md]: done

File: dbadvisor-core/src/main/java/com/github/embeditcz/dbadvisor/core/internal/analyzer/oracle/ForeignKeyAnalayzer.java (hunk 2 above)
Comment (mbocek) [reply dbadvisor-core/src/main/java/com/github/embeditcz/dbadvisor/core/internal/analyzer/oracle/ForeignKeyAnalayzer.java]: done

File: dbadvisor-core/src/main/java/com/github/embeditcz/dbadvisor/core/internal/analyzer/oracle/ForeignKeyAnalayzer.java (hunk 3 above)
Comment (mbocek) [reply dbadvisor-core/src/main/java/com/github/embeditcz/dbadvisor/core/internal/analyzer/oracle/ForeignKeyAnalayzer.java]: One point: query is part of comparator.
Maybe we should add datasource to comarator.

**Pull request merged**